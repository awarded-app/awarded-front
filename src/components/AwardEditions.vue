<template>
  <ul v-if="!editions">
    <li>Loading...</li>
  </ul>
  <ul v-else>
    <li v-if="editions.totalCount === 0">No editions!</li>
    <award-edition v-for="edition in editions.nodes" :key="edition.id" :edition="edition" />
  </ul>
</template>

<script>
import gql from 'graphql-tag'
import AwardEdition from '@/components/AwardEdition'
export default {
  name: 'AwardEditions',
  components: { AwardEdition },
  props: {
    awardId: {
      type: Number,
      default: null
    }
  },
  apollo: {
    editions: {
      query: gql`
        query editions($condition: EditionCondition) {
          editions(condition: $condition) {
            totalCount
            nodes {
              date
              name
              id
              poster
              award {
                nameLong
                nameShort
                logo
                id
              }
            }
          }
        }
      `,
      variables() {
        return {
          condition: {
            awardId: this.awardId
          }
        }
      },
      skip() {
        return this.awardId === null
      }
    }
  }
}
</script>

<style lang="scss" scoped></style>
