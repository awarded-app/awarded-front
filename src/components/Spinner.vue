<template functional>
  <hollow-dots-spinner
    :animation-duration="1000"
    :dot-size="15"
    :dots-num="3"
    color="#f2d024"
  />
</template>
