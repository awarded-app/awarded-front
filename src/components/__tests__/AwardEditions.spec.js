import { shallowMount, createLocalVue } from "@vue/test-utils";
import { customizer } from "./helper";
import AwardEditions from "../AwardEditions.vue";
import Spinner from "../Spinner.vue";
import { HollowDotsSpinner } from "epic-spinners";
import mergeWith from "lodash.mergewith";

// create an extended `Vue` constructor
const localVue = createLocalVue();
localVue.component("hollow-dots-spinner", HollowDotsSpinner);

function createWrapper(overrides) {
  const defaultMountingOptions = {
    propsData: {
      awardId: 0
    },
    mocks: {
      $apollo: {
        loading: false
      }
    },
    localVue
  };
  return shallowMount(
    AwardEditions,
    mergeWith(defaultMountingOptions, overrides, customizer)
  );
}
const editions = {
  totalCount: 3,
  nodes: [
    {
      date: "2017-02-26",
      name: "89th Academy Awards",
      id: 2,
      poster: "2017_Oscars_poster.jpg",
      __typename: "Edition"
    },
    {
      date: "2016-02-28",
      name: "88th Academy Awards",
      id: 8,
      poster:
        "b67e67830551ea34f84eb0edf7900236553b83045ca648f52711fe9838b7a681-699x1024.jpg",
      __typename: "Edition"
    },
    {
      date: "2015-02-22",
      name: "87th Academy Awards",
      id: 17,
      poster: "87thkeyart13.jpg",
      __typename: "Edition"
    }
  ],
  __typename: "EditionsConnection"
};

describe("AwardEditions.vue", () => {
  test("displays Spinner while query is loading", () => {
    const wrapper = createWrapper({
      mocks: {
        $apollo: {
          loading: true
        }
      }
    });

    expect(wrapper.find(Spinner).exists()).toBe(true);
  });
  test("does not display Spinner when loading is finished", () => {
    const wrapper = createWrapper();
    expect(wrapper.find(Spinner).exists()).toBe(false);
  });
  test("displayed awards correctly with query data", () => {
    const wrapper = createWrapper();
    wrapper.setData({
      editions
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
