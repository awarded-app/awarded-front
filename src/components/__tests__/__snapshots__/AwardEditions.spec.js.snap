// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwardEditions.vue displayed awards correctly with query data 1`] = `
<div>
  <div
    class="pl-8"
  >
    <!---->
     
    <award-edition-stub
      edition="[object Object]"
    />
    <award-edition-stub
      edition="[object Object]"
    />
    <award-edition-stub
      edition="[object Object]"
    />
  </div>
</div>
`;
